### ISSUE TITLE
bug: Fig holds too many ports

### ISSUE THREAD

**Issue by abstractalgo** [withfig/fig#2823]
### Checks

- [X] I have searched [github.com/withfig/fig/issues](https://github.com/withfig/fig/issues?q=) and there are no duplicates of my issue
- [X] I have run `fig doctor` in the affected terminal session
- [X] I have run `fig restart` and replicated the issue again

### Operating system

MacOS 13.6

### Expected behaviour

Not to slow down my computer

### Actual behaviour

Fig opens too many ports, and something seems off there:

<img width="831" alt="image" src="https://github.com/withfig/fig/assets/1355455/98d4a3e2-71d0-495e-8b26-076ba21c894c">

On the previous occasion, it had opened 24,000 ports and I noticed my computer slowing down. Only after killing the Fig process and restarting again did it go back to normal. Still, it continues to open and more ports over time, having way more ports even than Chrome browser, even though I have only ~5 terminals open.

### Steps to reproduce

Use Fig as normal and give it enough time to ramp up ports to multiple thousands.

### Environment

```yaml
fig-details:
  - 2.17.0
hardware-info:
  - model:
  - model-id:
  - chip-id: Apple M2 Pro
  - cores: 10
  - mem: 16.00 GB
os-info:
  - macOS 13.6.0 (22G120)
environment:
  - shell: /bin/zsh
  - terminal: iterm
  - cwd: /Users/dokanovic/code/aalgo
  - exe-path: /Users/dokanovic/.fig/bin/fig
  - install-method: unknown
  - env-vars:
    - FIGTERM_SESSION_ID: 6cbe958d-1661-4169-a3d8-a294b8e21f02
    - FIG_SET_PARENT_CHECK: 1
    - FIG_TERM: 2.16.0
    - PATH: /Users/dokanovic/.bun/bin:/Users/dokanovic/Library/pnpm:/opt/homebrew/opt/qt@5/bin:/Users/dokanovic/google-cloud-sdk/bin:/Users/dokanovic/.volta/bin:/Users/dokanovic/.pyenv/shims:/Users/dokanovic/.nvm/versions/node/v16.14.0/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Users/dokanovic/.fig/bin:/Users/dokanovic/.local/bin
    - SHELL: /bin/zsh
    - TERM: xterm-256color
    - __CFBundleIdentifier: com.googlecode.iterm2
    - FIG_PID: 12606
    - FIG_SET_PARENT: 6cbe958d-1661-4169-a3d8-a294b8e21f02
```

**Comment by mschrage** [comment]
Thank you for reporting this! We'll investigate what's going on here - definitely a bug!
